src/main/java/de/blablubbabc/paintball/thirdparty/PaintballPlaceholders.java

```diff
@@ -0,0 +1,91 @@
+package de.blablubbabc.paintball.thirdparty;
+
+import org.bukkit.entity.Player;
+
+import java.util.Map;
+
+import de.blablubbabc.paintball.Lobby;
+import de.blablubbabc.paintball.Paintball;
+import de.blablubbabc.paintball.Rank;
+import de.blablubbabc.paintball.statistics.arena.ArenaStat;
+import de.blablubbabc.paintball.statistics.general.GeneralStat;
+import de.blablubbabc.paintball.statistics.player.PlayerStat;
+import de.blablubbabc.paintball.statistics.player.PlayerStats;
+import me.clip.placeholderapi.expansion.PlaceholderExpansion;
+
+public class PaintballPlaceholders extends PlaceholderExpansion {
+    private final Paintball pb;
+
+    public PaintballPlaceholders(Paintball pb) {
+        // Do not use class variables for managers, they are dropped when paintball reloads
```

Comment:
I am not completely sure what this comment is about. But if it is about discouraging accessing the plugin instance via static variables: I don't think this is necessarily an issue, per-se.

However, it might be an issue if the extension is still registered even though the paintball plugin has already been disabled or reloaded. However, in this case I would also consider accessing the stale (unloaded) plugin instance via this member field to not be particularly safe.

Reply:
It was just a note to myself originally (when it was running as a separate plugin) to not store an instance of `PlayerManager` or `Stats` because those get reset when paintball reloads. Since it's now part of paintball and *should* be reloaded when paintball is reloaded, it probably doesn't matter one way or the other.

Reply:
It might make sense to remove this comment again, to not cause unnecessary confusion when I or someone else looks at it again later.